### DISCUSSION TITLE
Possible causes of initProducerId fail

### DISCUSSION THREAD

**Discussion by Psymboll** [spring-projects/spring-kafka#2617] [Q&A]
In our project that uses spring-kafka 2.6.5 we often encounter this exception: `org.springframework.transaction.CannotCreateTransactionException: Could not create Kafka transaction; nested exception is org.apache.kafka.common.errors.TimeoutException: Timeout expired after 60000milliseconds while awaiting InitProducerId...`
This happens right before the kafka producer in the service tries to send a message, so it has resulted in data loss in a number of occasions.
By looking at the logs, we witnessed the application trying to execute the initProducerId operation and then failing after having waited the 60s interval without any response. 
**What are the possible causes of this failure ?** 
We already tried to monitor the performance of the broker during the accidents and did not encounter any lag spike or abnormal use of memory which might explain why the request was not fulfilled in time. 

Thanks in advance, will be available to provide more information on our configuration if needed.

**Comment by garyrussell** [comment]
We are not Kafka experts here; best to ask such questions to a wider audience such as on Stack Overflow or one of their mailing lists.

However, most likely, perhaps, is there are not enough replicas to support transactions; by default 3 are required. Look at the server logs to confirm that.

See the `transaction.state.log.*` properties https://kafka.apache.org/documentation/#brokerconfigs_transaction.state.log.load.buffer.size
